src/components/NotificationModal.js

```diff
@@ -0,0 +1,144 @@
+
+
+let notificationTitle, notificationBody, projectID
+
+const mapStateToProps = (state) => ({
+  notificationProject: state.notificationProject,
+  notificationPayload: state.notificationPayload
+})
+
+const mapDispatchToProps = (dispatch) => ({
+  fetchNotificationProject(projectID) {
+    dispatch(fetchNotificationProject(projectID))
+  },
+})
+
+class NotificationModal extends Component {
+  constructor(props) {
+    super(props)
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props !== nextProps) {
+      const notification = this.props.notificationPayload
+      if (Platform.OS === 'ios'){
+        if (notification._alert !== undefined) {
+          notificationTitle = notification._alert.title
+          notificationBody = notification._alert.body
+          projectID = ( notification._data.data !== undefined ? notification._data.data.project_id : null)
+        }
+      } else {
+        notificationTitle = notification.title
+        notificationBody = notification.body
+        projectID = notification.project_id
+      }
+
+      if (this.isMobileProject(projectID)) {
+        this.props.fetchNotificationProject(projectID)
+      }
+    }
+  }
+
+  isMobileProject(projectID) {
+    return indexOf(projectID, MOBILE_PROJECTS) >= 0
+  }
+
+  handleClick() {
+    GoogleAnalytics.trackEvent('view from notification', this.props.notificationProject.display_name)
+
+    const zurl=`http://zooniverse.org/projects/${this.props.notificationProject.slug}`
+    Linking.canOpenURL(zurl).then(supported => {
+      if (supported) {
+        Linking.openURL(zurl);
+      } else {
+        Alert.alert(
+          'Error',
+          'Sorry, but it looks like you are unable to open the project in your default browser.',
+        )
+      }
+    })
+  }
+
+  render() {
+    const projectDisplay =
+      <View style={styles.projectContainer}>
+          <StyledText additionalStyles={[styles.subheader]} text={`About ${this.props.notificationProject.display_name}`} />
+          <StyledText additionalStyles={[styles.smallText]}  text={this.props.notificationProject.description} />
+
+          <View style={styles.container}>
+            <Button
+              handlePress={this.handleClick}
+              text={'Visit this project'} />
+          </View>
+      </View>
+
+    return (
+      <StyledModal
+        isVisible={this.props.isVisible}
+        setVisibility={this.props.setVisibility}>
+
+        <StyledText
+          textStyle='largeBold'
+          text={notificationTitle} />
+
+        <View style={styles.container}>
+          <StyledText
+            text={notificationBody} />
+        </View>
+
+        { (isEmpty(this.props.notificationProject) || (isEmpty(projectID))) ? null : projectDisplay }
+
+        <Button
+          handlePress={() => this.props.setVisibility(false)}
+          buttonStyle={'greyButton'}
```

Comment:
Final note: change this `buttonStyle` because now it looks like a disabled button.